src/metabase/driver/athena.clj

```diff
@@ -0,0 +1,324 @@
+      (->> result
+        (remove #(= (:name %) ""))
+        (remove #(string/starts-with? (:name %) "#")) ; remove comment
+        (distinct)) ; driver can return twice the partitioning fields
+      (recur
+       (conj result (describe-database->clj rs))
+       (.next rs)))))
+
+(defn- run-query
+  "Workaround for avoiding the usage of 'advance' jdbc feature that are not implemented by the driver yet.
+   Such as prepare statement"
+  [database query {:keys [read-fn]}]
+  (let [current-read-fn (or read-fn #(into [] (jdbc/result-set-seq % {:identifiers identity})))]
+    (log/infof "Running Athena query : '%s'..." query)
+    (with-open [conn (jdbc/get-connection (sql/db->jdbc-connection-spec database))]
+      (-> (.createStatement conn)
+          (.executeQuery query)
+          (current-read-fn)))))
+
+(defn- column->base-type [column-type]
+  ({:array      :type/*
+    :bigint     :type/BigInteger
+    :binary     :type/*
+    :boolean    :type/Boolean
+    :date       :type/Date
+    :decimal    :type/Decimal
+    :double     :type/Float
+    :float      :type/Float
+    :int        :type/Integer
+    :map        :type/*
+    :smallint   :type/Integer
+    :string     :type/Text
+    :struct     :type/*
+    :timestamp  :type/DateTime
+    :tinyint    :type/Integer
+    :uniontype  :type/*
+    :varchar    :type/Text} (keyword column-type)))
+
+(defn- connection-details->spec
+  "Create a database specification for an Athena database. DETAILS should include keys for `:user`,
+   `:password`, `:s3_staging_dir` `:log_path` and `region`"
+  [{:keys [user password s3_staging_dir log_path region] :as details}]
+  (assert (or user password s3_staging_dir log_path region))
+  {:classname "com.amazonaws.athena.jdbc.AthenaDriver"
+   :log_path log_path
+   :subprotocol "awsathena"
+   :subname (str "//athena." region ".amazonaws.com:443")
+   :user user
+   :password password
+   :s3_staging_dir s3_staging_dir})
+
+(defn- can-connect? [driver details]
+  (let [conn-details (-> details
+                         (ssh/include-ssh-tunnel)
+                         (connection-details->spec))]
+    (with-open [connection (jdbc/get-connection conn-details)]
+      (-> (.createStatement connection)
+          (.executeQuery "SELECT 1")
+          (jdbc/result-set-seq)
+          (#(= 1 (first (vals (first %)))))))))
+
+(defn- get-databases [driver database]
+  (->> (run-query database "SHOW DATABASES" {})
+       (remove #(= (:database_name %) "default"))
+       (map (fn [{:keys [database_name]}]
+              {:name database_name :schema nil}))
+       (set)))
+
+(defn- get-tables-from-db [driver db database]
+  (->> (run-query db (str "SHOW TABLES IN " (:name database)))
+       (map (fn [{:keys [tab_name]}]
+              {:name tab_name
+               :schema (:name database)}))))
+
+(defn- get-tables-from-dbs [driver db databases]
+  (->> databases
+       (map #(get-tables-from-db driver db %))
+       (flatten)
+       (set)))
```

Comment:
in `->` and `->>` you don't need to wrap single forms in `()`. The macros do that for you automatically. So you can just do `flatten` instead of `(flatten)`